src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1311,35 +1311,33 @@ protected override void GetItem(string path)
                     // If we want to retrieve the file streams, retrieve them.
                     if (retrieveStreams)
                     {
-                        if (!isContainer)
+                        foreach (string desiredStream in dynamicParameters.Stream)
                         {
-                            foreach (string desiredStream in dynamicParameters.Stream)
+                            // See that it matches the name specified
+                            WildcardPattern p = WildcardPattern.Get(desiredStream, WildcardOptions.IgnoreCase | WildcardOptions.CultureInvariant);
+                            bool foundStream = false;
+
+                            foreach (AlternateStreamData stream in AlternateDataStreamUtilities.GetStreams(result.FullName))
                             {
-                                // See that it matches the name specified
-                                WildcardPattern p = WildcardPattern.Get(desiredStream, WildcardOptions.IgnoreCase | WildcardOptions.CultureInvariant);
-                                bool foundStream = false;
+                                if (!p.IsMatch(stream.Stream))
+                                { continue; }
 
-                                foreach (AlternateStreamData stream in AlternateDataStreamUtilities.GetStreams(result.FullName))
-                                {
-                                    if (!p.IsMatch(stream.Stream)) { continue; }
+                                string outputPath = result.FullName + ":" + stream.Stream;
+                                WriteItemObject(stream, outputPath, isContainer);
+                                foundStream = true;
+                            }
 
-                                    string outputPath = result.FullName + ":" + stream.Stream;
-                                    WriteItemObject(stream, outputPath, isContainer);
-                                    foundStream = true;
-                                }
+                            if ((!WildcardPattern.ContainsWildcardCharacters(desiredStream)) && (!foundStream))
+                            {
+                                string errorMessage = StringUtil.Format(
+                                    FileSystemProviderStrings.AlternateDataStreamNotFound, desiredStream, result.FullName);
+                                Exception e = new FileNotFoundException(errorMessage, result.FullName);
 
-                                if ((!WildcardPattern.ContainsWildcardCharacters(desiredStream)) && (!foundStream))
```

Comment:
The behavior should be preserved as I commented in related issue.

Reply:
The behavior is preserved.  Catching the Win32Exception that would have originally been thrown on ERROR_HANDLE_EOF (down at line 8634) just prevents the Win32Exception from propagating up and being uncaught.  The code up here checks to see whether the returned List<AlternateDataStream> contains the desired stream and also checks the presence of wildcard characters.  If it doesn't find it and desiredStream doesn't contain wildcards, it branches into this terminating exception.

Reply:
Sorry, I skipped new indentation.